### PR TITLE
Feature/unstoppabledomains

### PR DESCRIPTION
**Pull request by JohnnyJumper**
## Description
Integration with Unstoppabledomains using @unstoppabledomains/resolution for resolving ens domain names as well as .zil and .crypto.

## Changes
Added UnstoppableResolution service that handles the domain resolution and error messages

## Steps to Test

* Go to send Assets page
* Choose assets to send (eth as an example)
* Type in recipient field any .eth .luxe .zil or .crypto domains
* press tab
* see the resolved domain's address if any or a corresponding error such as Domain is not registered

## Quality Assurance

- [x] The branch name is in lowercase-kebab-case with no prefix (unless it was created from Clubhouse)
- [ ] The base branch is develop or gau (no nested branches)
- [ ] This is related to a maximum of one Clubhouse story or GitHub issue
- [x] Types are safe (avoid TypeScript/TSLint features like any and disable, instead use more specific types)
- [x] If code is copied from existing directories, there is an explanation of why this is necessary in the description/changes, and all copying is done in separate commits to make them easy to filter out

### CODE REVIEW

**Comment by LuckyLuker** [comment]
[ch4826] - This is for our internal project management.

File: common/v2/services/UnstoppableService/index.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  public parseResolutionError(err: ResolutionError, callback: any, params?: any) {
+    const setResolutionError = callback;
+    const { ticker, name } = params;
+    setResolutionError({
+      code: err.code,
+      params: { $domain: name, $recordName: ticker, $currencyTicker: ticker }
+    });
+  }
+}
```
Comment (vilsbole) [inline common/v2/services/UnstoppableService/index.ts]: There seems to be no reason to execute the callback here.
Prefer a pure function that returns the formatted object, and execute the side-effect directly within the component.

**Comment by sudoryan** [comment]
This latest push should resolve all the feedback with the exception of the missing functional test for the DomainStatus component. @JohnnyJumper will write and push the missing test early next week.

File: common/v2/components/DomainStatus.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      <>
+        <Spinner /> {translate('LOADING_ENS_ADDRESS')}
+      </>
+    );
+  }
+  if (!isError && !resolutionError && isHexString(rawAddress)) {
+    return <>{`${translateRaw('SEND_ASSETS_DID_RESOLVE')}: ${rawAddress}`}</>;
+  }
+  if (resolutionError && UnstoppableResolution.isValidDomain(rawAddress)) {
+    return <>{translateRaw(resolutionError.code.toUpperCase(), resolutionError.params)}</>;
```
[the remark this answers is not in the document]
Comment (JohnnyJumper) [reply common/v2/components/DomainStatus.tsx]: I removed the test and translations. For translations, I favored the ResolutionError.message that we set for the error before being thrown. That is the English language only.
For the test, I couldn't find a way how to mock the method without exporting its internals which is not very reliable and might become a breaking change for you in the future.

[the remark this answers is not in the document]
Comment (vilsbole) [reply common/v2/components/DomainStatus.tsx]: All is good. Thanks!

**Pull request merged**